shell/common/node_bindings.cc

```diff
@@ -280,6 +280,10 @@ void SetNodeOptions(base::Environment* env) {
 
     // overwrite new NODE_OPTIONS without unsupported variables
     env->SetVar("NODE_OPTIONS", options);
+    // https://github.com/electron/electron/issues/27551
+#if defined(OS_WIN)
```

Comment:
on further thought - the root issue seems to be 
> Electron is setting updated NODE_OPTIONS using SetEnvironmentVariable and Node is retrieving them via getenv.

I think instead here we should remove this platform specificity and instead also call

```cpp
setenv("NODE_OPTIONS", options);
```

Reply:
We still need to call this version of the API on Windows as it's supposed to be safer than plain `setenv`, according to MSDN:
> Creates, modifies, or removes environment variables. These are versions of _putenv, _wputenv but have security enhancements, as described in Security Features in the CRT.

Reply:
replaced by @deepak1556 suggestion